### PR TITLE
Wanderer: Add dialog explaining Wanderer to Hai translation app

### PR DESCRIPTION
**Pull request by williaji**
## Summary
As the game currently plays, every Wanderer can speak Hai and be understood by the player. This adds a brief addition to the initial meeting with Rek to explain that they have a machine translation application that will translate to Hai. 

This has been discussed in https://github.com/endless-sky/endless-sky/pull/6809

## Save File
This save file can be used to play through the new mission content:
{{attach a save file that allows people to easily test your added mission content or see your new in-game art}}
[Vander Lost.txt](https://github.com/endless-sky/endless-sky/files/9569932/Vander.Lost.txt)

## Testing Done
Tested in game. Content renders as expected.
Wanderer Translator Application Dialogue

### CODE REVIEW

**Comment by reticent-rem** [comment]
That's getting to be a pretty big block of text. Could we add a choice at the bottom like `Does every Wanderer speak Hai?` and put the explanation behind that?

**Comment by MasterOfGrey** [comment]
So, short answer: No

Longer answer: One of the great under-used ways to provide limitations which raise the stakes on making solutions in the Wanderer campaign is the possibility that not every wanderer might be able to understand the language you have available. This is something that has been floated for a couple of points I've seen discussed in the plans to push the story forward.
For that reason alone I would be against this.
However, in addition to that, there's content being wrapped up for addition very soon which changes a few things, and one of the consequences of that is that soon after there'll be new content with ways to have some exchanges with the Wanderers without initiating official first contact, and without being able to be understood except through good old fashioned pointing and gesturing.
What you have here is one step from a universal translator - which aside from messing with the laid out plan for imminent content, is also something that has been explicitly rejected in the past. Never mind the fact that in this case it's essentially a magic rune of translation.

I appreciate the effort here, but things are heading the opposite direction at present.

**Comment by roadrunner56** [comment]
@Hecter94 I'll hold off on reviewing this until it's decided whether this PR's concept is good or not.

**Comment by williaji** [comment]
I'll close the PR though at least part of the rationale confuses me. This exchange occurs after first contact and at the point where you have a  translation device in the current story line, so exchanges before this point are not affected. This is *not* a universal translator. It translates Wanderer to Hai just as the player has a Hai to Human  translator; it's not different to what the player has.

**Comment by williaji** [comment]
On reflection, I've decided to re-submit his because the justifications provided don't make sense to me. Obviously, Grey can decide to reject it, but I'd want to see if anyone else chose to weigh in on the merits of the idea. I think it at least deserves a fuller discussion.

**Comment by Saugia** [comment]
That's okay, but be sure to respect Grey's standpoints as he is basically the lead for Hai content, and if he feels something doesn't fit in, that will remain a conflict unless it's worked around.

**Comment by williaji** [comment]
> That's okay, but be sure to respect Grey's standpoints as he is basically the lead for Hai content, and if he feels something doesn't fit in, that will remain a conflict unless it's worked around.

l respect Grey's view. I specifically said it was up to him to reject it. My concerns are:

1. It's being rejected because it might foreclose some content that's been kicked around as an idea. I assume that's correct as I have no knowledge. But, this concept was presented months ago in the Spera Anatrusk outfitter PR and the the issue of it blocking other content was never raised. What *was* raised was the fact that there is no mention of such an app. So, I submitted this to address that issue. It seems there should either be discussion about the best way forward and this PR seems to be the thing the raises it. If it's decided by the powers that be, fine.  But as far as I know, those other ideas are not visible and are not currently submitted for implementation. 
2.  It was objected to because "there'll be new content with ways to have some exchanges with the Wanderers without initiating official first contact". This dialogue occurs *after* first contact. It occurs *after* you have a translation device that allows you to communicate. So I do not understand how this change can have any impact on that.
3. It was objected to because "What you have here is one step from a universal translator." That is not correct. The application translates from Wanderer to Hai. To allow Wanderers who come across Hai to be able to communicate in a rudimentary fashion. It is, in fact, a less sophisticated version of the device the player already has, but handles the Wanderer side of the conversation. We know that they have captured Unfettered troops, so their must be occasional encounters.
4. There is no discussion of the elephant in the room: the existing story line allows us to talk to any Wanderer on any planet using the Hai language. Dock workers, scientists, farm laborers, children and their teachers are all apparently able to speak the Hai language for the benefit our our translator. To me that is a far bigger strain on my suspension of disbelief. And, if we are going to move to a model where not all Wanderers can communicate with the player, we have a lot of editing to do on the existing content.

**Comment by Saugia** [comment]
You may be overlooking that first contact is getting a huge addition in content centered around it, both before and after, in PR https://github.com/endless-sky/endless-sky/pull/4952. That's currently the priority and it will change and/or add quite a bit around the hai's early content.

**Comment by MasterOfGrey** [comment]
> l respect Grey's view. I specifically said it was up to him to reject it. My concerns are:

To start, thank you for getting into the details of it. I appreciate it.

> 1. It's being rejected because it might foreclose some content that's been kicked around as an idea. I assume that's correct as I have no knowledge. But, this concept was presented months ago in the Spera Anatrusk outfitter PR and the the issue of it blocking other content was never raised. What _was_ raised was the fact that there is no mention of such an app. So, I submitted this to address that issue. It seems there should either be discussion about the best way forward and this PR seems to be the thing the raises it. If it's decided by the powers that be, fine.  But as far as I know, those other ideas are not visible and are not currently submitted for implementation.

In this case it's not so much "kicked around as an idea" as it is "planned to address a very specific concern that Amazinite has because of how 4952 changes pre-established expectations". It is content that's not fully defined yet, but it also is content that's definitely happening. I regret that I didn't pick up on things when I skimmed the Spera Anatrusk missions, but since it'd been assigned to Amazinite I hadn't given it a close look.

> 2. It was objected to because "there'll be new content with ways to have some exchanges with the Wanderers without initiating official first contact". This dialogue occurs _after_ first contact. It occurs _after_ you have a translation device that allows you to communicate. So I do not understand how this change can have any impact on that.

If it's something ubiquitously available, then it stretches disbelief that there would be any reasonable difficulties with first contact to start with. You wouldn't need to have landed on one specific world and met one specific wanderer for things to trigger.

> 3. It was objected to because "What you have here is one step from a universal translator." That is not correct. The application translates from Wanderer to Hai. To allow Wanderers who come across Hai to be able to communicate in a rudimentary fashion. It is, in fact, a less sophisticated version of the device the player already has, but handles the Wanderer side of the conversation. We know that they have captured Unfettered troops, so their must be occasional encounters.

It is, and this is why:
The premise is that alien languages are not as simple as just translating the sounds. When we translate between human languages we share the basis that while we're looking at each other and hearing the translation, we still have a common frame of reference for body language and facial expressions.
The translator the Hai give you is sophisticated in that it uses cameras and holograms etc. to read those non-verbal elements and also communicate non-verbal elements back, because *we don't* have that common frame of reference with the Hai. (Certainly many Hai have lived near humans enough to bridge that gap, but intrinsically it's not always there.)
When you talk to Rek in Hai you're talking to a linguist and a diplomat that has also learnt to mimic the essentials of those in order to communicate effectively. If anyone else can just use a verbal translation app to go from Wanderer to Hai, then as a character he's essentially pointless.

Body language can already change the meaning of things in our conversations to a large degree - the potential for miscommunication is extreme using something like this if people expect to get understood by it. And if the potential for serious miscommunication is not in-built into the use of this device for basically any serious conversation- then it's one step from being a universal translator.
Which of course, does lead into your next point:

> 4. There is no discussion of the elephant in the room: the existing story line allows us to talk to any Wanderer on any planet using the Hai language. Dock workers, scientists, farm laborers, children and their teachers are all apparently able to speak the Hai language for the benefit our our translator. To me that is a far bigger strain on my suspension of disbelief. And, if we are going to move to a model where not all Wanderers can communicate with the player, we have a lot of editing to do on the existing content.

Yes, this is something that should be changed. The Wanderer storyline is essentially 1/2 to 2/3 of a first draft. That's it. It's nowhere near a finished product, and yeah, the expectation is that issues like this will need to be properly addressed when more serious development on it takes place.


Hopefully that provides you with some clarity on the matter.

**Comment by williaji** [comment]
> In this case it's not so much "kicked around as an idea" as it is "planned to address a very specific concern that Amazinite has because of how 4952 changes pre-established expectations". It is content that's not fully defined yet, but it also is content that's definitely happening. 

So, does this mean that we should foreclose any new Wanderer PRs? Or should we create a draft PR and collect possible content with the understanding that it will likely have to be modified to integrate with Hai Reveal and any other changes that have been done? This is an honest question for me. I would like to contribute but I'm new (and slow) and am still learning the dynamics here. I have some missions sketched out to expand the early and middle phases. I've floated them on Discord but they get lost in the thread. I've not found anyone who's really driving the Wanderer story line so I can't plug into a larger story. I've tried to find targeted, stand alone content that I can do to the fill in gaps (Spera Antrusk). The subtext I'm getting is that Wanderer content isn't wanted just now.

> If it's something ubiquitously available, then it stretches disbelief that there would be any reasonable difficulties with first contact to start with. 

This would be true if the player understood Hai. If the player does not understand Hai, having the equivalent of a tourist guide translator (Where is the bathroom? How much does it cost?) that goes from one language you don't understand to another language you don't understand changes nothing regardless of how widely it's distributed. 

> It is, and this is why: The premise is that alien languages are not as simple as just translating the sounds....

I think we are talking past each other. I see this translator as an explanation for how the player can understand what's going on well enough to trade and do jobs. As I said above, a tourist phrase book, at best. As you say, Rek is a linguist and a diplomat. He serves as a bridge to Wanderer culture. As such, he is indispensable. No translator is going to be able to explain the Wander relationship to the Pug or their willingness to cede territory to the Unfettered.  
 
> Yes, this is something that should be changed. The Wanderer storyline is essentially 1/2 to 2/3 of a first draft. That's it. It's nowhere near a finished product, and yeah, the expectation is that issues like this will need to be properly addressed when more serious development on it takes place.

And, we are back to the main issue: do we want to collect potential Wanderer content do we want to put it all on hold?

**Comment by MasterOfGrey** [comment]
> So, does this mean that we should foreclose any new Wanderer PRs? Or should we create a draft PR and collect possible content with the understanding that it will likely have to be modified to integrate with Hai Reveal and any other changes that have been done? This is an honest question for me. I would like to contribute but I'm new (and slow) and am still learning the dynamics here. I have some missions sketched out to expand the early and middle phases. I've floated them on Discord but they get lost in the thread. I've not found anyone who's really driving the Wanderer story line so I can't plug into a larger story. I've tried to find targeted, stand alone content that I can do to the fill in gaps (Spera Antrusk). The subtext I'm getting is that Wanderer content isn't wanted just now.

I know Amazinite intends to get to Spera Anatrusk after Hai Reveal (which is imminent), so I wouldn't foreclose it. I would be prepared for changes to be asked of it in a little while though. In general it's an area of development that has been left to sit because there were more questions than answers, and I think it's going to be a gnarly one.
There is a sibling server to the main discord though for discussing it: https://discord.gg/MAs8sYuqKt
They made some progress laying out some possible plans, but I think it stalled.

I do encourage at least floating the ideas for expanding the early plot of wanderers. I don't know what might get picked up yet, but it certainly needs it.

> This would be true if the player understood Hai. If the player does not understand Hai, having the equivalent of a tourist guide translator (Where is the bathroom? How much does it cost?) that goes from one language you don't understand to another language you don't understand changes nothing regardless of how widely it's distributed.

That almost creates a larger problem for the actual plot, in that you then have to decide at every conversation whether or not it's adequate for the task or whether you need to recruit a proper interpreter.
At least for things like jobs etc. you could argue that those are all picked up and accepted via some sort of computer ticketing system.

> I think we are talking past each other. I see this translator as an explanation for how the player can understand what's going on well enough to trade and do jobs. As I said above, a tourist phrase book, at best. As you say, Rek is a linguist and a diplomat. He serves as a bridge to Wanderer culture. As such, he is indispensable. No translator is going to be able to explain the Wander relationship to the Pug or their willingness to cede territory to the Unfettered.

See above, with some minor text-based plugins to your systems you could argue those are done without any person-person contact at all.

> And, we are back to the main issue: do we want to collect potential Wanderer content do we want to put it all on hold?

I'd say check out the server just for it and float most content ideas there. 
It's an awkward case that wanderers is probably not a priority for the next 6-8 months, but the ideas need work to line up still. Previously there's been a bigger shortage of ideas than willingness, so the discussions are worth having in that forum if you want to dive into this.

**Comment by williaji** [comment]
> At least for things like jobs etc. you could argue that those are all picked up and accepted via some sort of computer ticketing system.

This begs the question of how the player can use the computer ticketing system that expresses it's output in a Wanderer language representation.  Something or someone has to be putting this information in a human comprehensible format. I have accidentally set my computer to a non-latin character set and it was exciting getting it back.

It seems to me that the primary thing you want to communicate the inherent difficulty of cross-species communication. If that's the case, a basic translation app could actually be a mechanism for that. If the jobs are re-cast with a degree of confusion in a dialog with phrases like, "You ask them to repeat the quantity" or "You think they said" you would provide the ambiguity. We could follow up with a small selection of jobs that send you to the wrong location or pay a different amount or have more people who up show up to board, etc. This would both explain how you are able to communicate across Wanderer space and reinforce the fact that it's duct tape and string.

**Comment by pilover100** [comment]
As Grey has said the wanderer server has stalled for the moment (No one I know of is currently in a position to tackle such a big project of reworking the wanderers), but floating ideas up in the dedicated server and letting them stay there (stick them into #story-creation if you don't want it to get lost) means that we can then consider them in the future. But yes, there are language issues. The first contact even shows that with a dedicated wanderer being needed to translate the leader's words into the hai language. 

I don't think though that a simple sticker is the solution here though, it feels rather "tacked on" and not integrated into the story. And whilst the hai translator isn't perfect, the hai have had hundreds of years potentially to perfect it as opposed to the wanderer's 20 years or so with the Uhai, that is a big difference in time. We humans only get away with things like google translate because we are all physically capable of the same sets of sounds and we ignore things like tone in the translation, an interspecies translator is going to need a lot more than that to work properly (e.g. sounds and gestures might be both *critical* to the meaning of a word).

So yes, whilst the wanderers need something/someone to translate into hai, a universal digital phrase book doesn't seem elegant... and if they're doing that why bother with the hai intermediary long term? just create a wanderer-human device. What I feel needs to happen is seeing more of the wanderers *failing* to communicate rather than just another glazed over language barrier.

(apologies for the large text wall btw)

**Comment by williaji** [comment]
> So yes, whilst the wanderers need something/someone to translate into hai, a universal digital phrase book doesn't seem elegant... and if they're doing that why bother with the hai intermediary long term? just create a wanderer-human device. What I feel needs to happen is seeing more of the wanderers _failing_ to communicate rather than just another glazed over language barrier.

The Wanderers have a clear use for a rudimentary translation device to allow them to communicate with Unfettered Hai who end up stranded and captured. They have no reason to create a human language device because they literally only know one human and that doesn't justify the effort.

I'm intrigued by the idea that having a phrase book to explain being able to use the job board is inelegant or tacked on. Having it here, at the beginning of our interaction with Rek integrates it at the beginning of the story line. Only experienced players will know it's retro-fitted. From my view it's more elegant that the current situation where everyone seems to speak Hai.

There is a fundamental dilemma in this discussion: on the one hand the view seems to be that we want to add a lot of friction to communicating with the Wanderers. On the other hand, people will abandon Wanderer space if they can't trade or do jobs (who spends time in the Graveyard?). So how do we thread this needle? 

I've not yet seen a proposal to explain how we allow people to participate in the Wanderer economy without language. To say "it's on the computer" is a fudge and makes everything appear seamless (as it does now) which is ostensibly what we want to avoid. The translation app can be the mechanism by which you illustrate the friction by having the jobs go wonky on occasion.

**Comment by Anarchist2** [comment]
I find it hardly a stretch of the imagination to believe that there are a dozen or so Wanderers terse in the Hai language on each planet, with each of them able to translate Wanderer jobs for you.

> To say "it's on the computer" is a fudge and makes everything appear seamless (as it does now) which is ostensibly what we want to avoid. The translation app can be the mechanism by which you illustrate the friction by having the jobs go wonky on occasion.

This is just saying, "Handwaving it with a computer is dumb. We should instead handwave it with a computer!"

**Comment by Zireael07** [comment]
@Anarchist2 No, what they're saying is "handwaving in the background is dumb, let's have at least a basic explanation"

File: data/wanderer/wanderers start.txt [hunk 1]
```diff
@@ -392,6 +392,7 @@ mission "Wanderers Conversation"
...
 			`	Rek is slightly larger than most of the Wanderers you have seen, with muted white and brown plumage and a calm, dignified bearing. "It is very [rare, exceptional] in our history that we have to [deal with, cope with] other species," he says, "and now we find ourselves dealing with two at once." He is speaking the Hai language, and the translation device does its best to render his words in your own language.`
+			`	Before you can ask a question, Rek holds out a thin square of metal covered in blocks of triangular marks that remind you of cunieform writing from ancient Earth. It easily grips your fingers as you take it from him. "Only a few of us have [studied, learned] the Hai language," he explains. "But, like your [box, device], we have an application that can be accessed at all our spaceports that will do a [rudimentary, basic] job sufficient for trade. This [badge, label] will [tell, inform] the workers how to access it. He mimes pressing his hand to his chest. When you do the same, the metal adheres firmly. You notice light marks on Rek's own clothing where his talons have pressed the fabric. They are similar to the writing on the badge.`
```
Comment (tibetiroka) [inline data/wanderer/wanderers start.txt:395]:
```suggestion
			`	Before you can ask a question, Rek holds out a thin square of metal covered in blocks of triangular marks that remind you of cuneiform writing from ancient Earth. It easily grips your fingers as you take it from him. "Only a few of us have [studied, learned] the Hai language," he explains. "But, like your [box, device], we have an application that can be accessed at all our spaceports that will do a [rudimentary, basic] job sufficient for trade. This [badge, label] will [tell, inform] the workers how to access it. He mimes pressing his hand to his chest. When you do the same, the metal adheres firmly. You notice light marks on Rek's own clothing where his talons have pressed the fabric. They are similar to the writing on the badge.`
```

**Review by tibetiroka — CHANGES_REQUESTED** [review]

**Comment by bene-dictator** [comment]
Will this be superceded by #8487 and #8488?

**Comment by Amazinite** [comment]
I'm not a fan of the idea of introducing another translation device to the Wanderer storyline, however rudimentary, on top of the existing Hai translation device. In some sense it feels like a lazy approach to solving the problem of how the player communicates with an alien faction. At the very least the existing translation device has a degree of plausibility to it existing.

The Hai have had close contact with humanity for some time now, but it's doubtful that all Hai know English (or whatever language you imagine humans to be speaking in the 31st century), or that many humans know Hai. The Hai are also a long lived and technologically advanced civilization. Given this dynamic, it isn't impossible to think that someone among the Hai would develop a translation device to communicate between Hai and English.

The Wanderers have also been in contact with the Unfettered for a similarly long amount of time. Just as the Hai have learned English, the idea that there are Wanderers who have learned Hai from captured Unfettered is not impossible to believe. Therefore, we have a connection by which you can use the Hai translation device to speak with *some* Wanderers.

But where does this "translation app" for the Wanderers come from? The Wanderers have no real need for such technology. Their only outside interactions for thousands of years have been with the Unfettered in the past one to two hundred, and it has been a hostile interaction at that. Yet they somehow have something set up at all spaceports that allows you to communicate from English to Hai to Wanderer for the sake of basic communication? Why would their spaceports have technology for translating Hai? A vast majority of the Wanderer worlds likely have had a single Hai on them up until the events of the campaign. And if this translation app isn't translating Hai, then what could it plausibly be translating that allows you to interact with most Wanderers?

Now maybe it could be said that it's implausible that you be offered so many jobs on Wanderer worlds if you can't effectively communicate with most of them, but I feel like it make more sense to hand wave that for the sake of gameplay rather than to ham-fist in some sort of flimsy explanation as to why you're able to get along so well in Wanderer society.

If we think there's a disconnect between what is stated or implied and what we experience in the story, the stated/implied information being that most Wanderers don't know Hai and the experience being that a lack of effective communication somehow doesn't hinder us in any way, then I think the correct approach is not to give an explanation for why communication is easy, but to change the story so that communication is harder. This is actually something that the Gegno storyline will be dealing with, and outside of the single translator you meet during the Bunrodea first contact, it will be a similar case in my planned Bunrodea content. That being that there will be some factions that we simply cannot communicate with through spoken language. I think it would do us well to have the Wanderer campaign include some of these ideas as well, instead of having most factions in the game be capable of conversing with us most of the time. 

TL;DR I don't want any sort of device that makes communication with Wanderers easier, I want communication with Wanderers to become harder for the player within the story.
